app/src/main/java/net/gsantner/markor/format/general/CommonTextActions.java

```diff
@@ -204,6 +215,57 @@ public void onClick(DialogInterface dialog, int selectedColor, Integer[] allColo
         return false;
     }
 
+    protected void runIndentLines(Boolean deIndent) {
+
+        Editable text = _hlEditor.getText();
+
+        int[] selection = getSelection();
+        int selectionStart = selection[0];
+        int selectionEnd = selection[1];
+
+        int lineStart = StringUtils.getLineStart(text, selectionStart);
+
+        char[] tabChars = new char[_tabWidth];
+        Arrays.fill(tabChars, ' ');
+        String tabString = new String(tabChars);
+
+        while (lineStart <= selectionEnd) {
+
+            if (deIndent) {
+                int textStart = StringUtils.getNextNonWhitespace(text, lineStart, selectionEnd);
+                int spaceCount = textStart - lineStart;
+                int delCount = Math.min(_tabWidth, spaceCount);
+                int delEnd = lineStart + delCount;
+                if (delCount > 0 && delEnd < text.length()) {
+                    text.delete(lineStart, delEnd);
+                    selectionEnd -= delCount;
+                }
+            }
+            else {
+                text.insert(lineStart, tabString);
+                selectionEnd += _tabWidth;
+            }
+
+            text = _hlEditor.getText();
+            // Get next line
+            lineStart = StringUtils.getLineEnd(text, lineStart, selectionEnd) + 1;
+        }
+    }
```

Comment:
Moved `runIndentLines` routine from TextActions to CommonTextActions and hooked it up to the indent and de-indent keyboard shortcut.